_automating-workflows/workflow-steps.md

```diff
@@ -0,0 +1,58 @@
+|`user_inputs`	|Object	|Optional	|A key-value map of inputs supported by the corresponding API for this specific step. Some inputs are required for an API, while others are optional. Required inputs may be specified here if known, or in the `previous_node_inputs field`.  The [Get Workflow Steps API]({{site.url}}{{site.baseurl}}/automating-workflows/api/get-workflow-steps/) identifies required inputs and step outputs. <br> Substitutions are supported in string values, lists of strings, and maps with string values. The pattern `{% raw %}${{previous_step_id.output_key}}{% endraw %}` will be replaced by the value in the previous step's output with the given key.  For example, if a parameter map in the user inputs includes a key `embedding_model_id` with a value `{% raw %}${{deploy_embedding_model.model_id}}{% endraw %}`,  then the `model_id` output of the `deploy_embedding_model` step will be substituted here. This performs a similar function to the `previous_node_input` map, but is not checked in validation and does not automatically infer edges. <br>In some cases, you can include [additional inputs](#additional-fields) in this field.	|
+
+## Workflow step types
+
+The following table lists the workflow step types. The fields under `user_inputs` for these steps correspond directly to the linked APIs.
+
+|Step type	|Corresponding API	|Description	|
+|---	|---	|---	|
+|`noop`	|No API	|A no-op (no operation) step that does nothing. It may be useful in some cases for synchronizing parallel steps.	|
+|`create_connector`	|[Create Connector]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/connector-apis/create-connector/)	|Creates a connector to a model hosted on a third-party platform.	|
+|`delete_connector`	|[Delete Connector]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/connector-apis/delete-connector/)	|Deletes a connector to a model hosted on a third-party platform.	|
+|`register_remote_model`	|[Register Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/)	|Registers a model hosted on a third-party platform. If the `user_inputs` field contains a `deploy` key that is set to `true`, also deploys the model.	|
+|`register_model_group`	|[Register Model Group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-group-apis/register-model-group/)	|Registers a model group. The model group will be deleted automatically once no model is present in the group.	|
```

Comment:
@joshpalis should we also add local models workflow steps here?

Reply:
yes we have `RegisterLocalPretrainedModel`, `RegisterLocalCustomModel`, and `RegisterLocalSparseEncodingModel` steps

Reply:
Can you mention a one liner for each step here so that @kolchfa-aws can update the doc?

Reply:
@joshpalis I updated the doc with these 3 steps. I assume they also have an option of deploying when `user_inputs` contains `deploy`=`true`. Please review. Thanks!